packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -116,7 +116,18 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
     );
 
     return z.object({
-      visitStartDate: z.date(),
+      visitStartDate: z.date().refine(
+        (value) =>
+          displayVisitStopDateTimeFields
+            ? true
+            : dayjs(value).isBefore(dayjs(), 'day') || dayjs(value).isSame(dayjs(), 'day'),
+        t('invalidVisitStartDate', 'Start date needs to be on or before {{firstEncounterDatetime}}', {
+          firstEncounterDatetime: new Date().toLocaleString(),
```

Comment:
Please use `formatDatetime()` from esm-framework to ensure date display is consistent with the rest of the app. (`toLocaleString()` is locale-aware, but it is not necessarily aware of the applications locale).